src/parser/psf-parser.js

```diff
@@ -0,0 +1,167 @@
+                    chainname = atominfo[1].trim();
+                    if (residins_regex.test(atominfo[2])){
+                        var matches = residins_regex.exec(atominfo[2]);
+                        resno = parseInt(matches[0]);
+                        inscode = matches[1];
+                    }else {
+                        resno = parseInt(atominfo[2]);
+                        inscode = '';
+                    }
+                    resname = atominfo[3].trim();
+                    atomname = atominfo[4].trim();
+                    element = atominfo[5].trim();
+
+                    atomStore.growIfFull();
+                    atomStore.atomTypeId[idx] = atomMap.add(atomname, element);
+                    atomStore.serial[idx] = serial;
+                    atomStore.bfactor[idx] = 0;
+                    atomStore.altloc[idx] = NaN;
+                    atomStore.occupancy[idx] = 0;
+
+                    sb.addAtom(modelIdx, chainname, resname, resno, 0, undefined, inscode);
+
+                    serialDict[serial] = idx;
+                    idx += 1;
+                }else if (section === 'bonds'){
+                    var bondlist = line.split(' ');
+                    var bondIndex = {};
+                    for (j = 0; j < bondlist.length; j=j+2){
+                        var from = parseInt(bondlist[j]) - 1;
+                        var to = parseInt(bondlist[j+1]) - 1;
+
+                        if( from < to ){
+                            ap1.index = from;
+                            ap2.index = to;
+                        }else{
+                            ap1.index = to;
+                            ap2.index = from;
+                        }
+
+                        // interpret records where a bond is given multiple times as double/triple bonds
+                        if( bondIndex[ to ] !== undefined ){
+                            s.bondStore.bondOrder[ bondIndex[ to ] ] += 1;
+                        }else{
+                            var hash = ap1.index + "|" + ap2.index;
+                            if( bondDict[ hash ] === undefined ){
+                                bondDict[ hash ] = true;
+                                bondIndex[ to ] = s.bondStore.count;
+                                s.bondStore.addBond( ap1, ap2, 1 );  // start/assume with single bond
+                            }
+                        }
+                    }
+                    s.hasConnect = true;
+                }
+
+                if (line.includes('!NATOM')) {
+                    section = 'atoms'
+                } else if (line.includes('!NBOND')) {
+                    section = 'bonds'
+                }
+            }
+        }
+
+        this.streamer.eachChunkOfLines( function( lines/*, chunkNo, chunkCount*/ ){
+            _parseChunkOfLines( 0, lines.length, lines );
+        } );
+
+        sb.finalize();
+
+        if( unitcellDict.a !== undefined ){
```

Comment:
there is no unitcell data, right? However, if there is data about the simulation box, that can be put here, for example as in the gro parser

```javascript
s.unitcell = new Unitcell(
  boxes[ 0 ][ 0 ], boxes[ 0 ][ 4 ], boxes[ 0 ][ 8 ],
  90, 90, 90, "P 1"
);
```